### ISSUE TITLE
GNS3 VM based on Alpine Linux

### ISSUE THREAD

**Issue by grossmj** [GNS3/gns3-vm#160]
I am considering having the GNS3 VM based on Alpine Linux. Any comments?

**Comment by jnatschev** [comment]
no opinion one way or another. alpine will more than likely be much slimmer.

**Comment by grossmj** [comment]
> likely be much slimmer.

Yes that's one of the reason :)

**Comment by b-ehlers** [comment]
First of all I think, that a slimmer GNS3VM would be nice. During the v2.2 releases the GNS3VM has more than doubled its size, while there was no such increase during v2.0 and v2.1. I doubt, that the only reason is Ubuntu, there must be also some other reasons. As long as these reasons are unknown, I don't know if switching to Alpine solved to base problem.

Size of the compressed image for VMware Workstation:

| Version | Size [MB] | Comment |
|--|--|--|
| 2.0.0 | 316 |
| 2.1.0 | 320 |
| 2.1.21 | 323 |
| 2.2.0 | 502 |
| 2.2.8 | 549 |
| 2.2.17 | 631 |
| 2.2.18 | 687 | Switch to Focal |
| 2.2.21 | 939 |
| 2.2.27 | 962 |
| 2.2.28 | 1180 |

Concerning Alpine I like to make the following comments:

* Alpine supports the main repository of each release for 2 years, while the community repository is only supported until next stable release (6 months). As some packages like QEMU and Docker are in the community repository, a VM based on Alpine has to upgrade its base OS every 6 month.
* Alpine uses very new software versions, that may have more bugs, than the older packages in Ubuntu.
* Since Alpine v3.14 GNS3 is unmaintained, see <https://git.alpinelinux.org/aports/commit/?id=b45777e14fcfefc047b4a46d2489fc575583fe68>. So the GNS3 devs must be prepaired to solve some issues before GNS3 can be used in Alpine.
* Alpine doesn't support the 32-bit glibc, needed for IOU.

**Comment by grossmj** [comment]
> First of all I think, that a slimmer GNS3VM would be nice. During the v2.2 releases the GNS3VM has more than doubled its size, while there was no such increase during v2.0 and v2.1.

Yes, it is annoying that the VM is getting bigger and bigger, that's why I am looking for an alternative. Right now we use the regular Ubuntu Focal LTS server version and then try to remove what's not needed https://github.com/GNS3/gns3-vm/blob/focal-stable/scripts/cleaner.sh however it looks like this strategy isn't as efficient as before...

To be honest, I am not sure switching to Alpine Linux is the solution. We would probably manage to bring down the size of the VM however this would create other problems in addition of the one you mentioned, like:

- We will not be able to install dependencies like uBridge etc. from our PPA
- Probably other issues with creating the VMs, we would have to update all the scripts including Packer json files.
- As you pointed out Alpine doesn't support the 32-bit glibc, needed for IOU

What I have in mind now would be to use a Ubuntu cloud image from https://cloud-images.ubuntu.com/ possibly even the minimal version. I am waiting for the next LTS release do it.

**Comment by grossmj** [comment]
I managed to build using the latest Ubuntu 26.04 LTS while keeping the size down. We could switch to using Debian cloud images that are smaller than the Ubuntu ones, if we need someday.
